index.js

```diff
@@ -255,9 +255,13 @@ function solveChallenge (options, response, body) {
 
   // Prevent reusing the headers object to simplify unit testing.
   options.headers = Object.assign({}, options.headers);
-  // Use the original uri as the referer and to construct the answer url.
+  // Use the original uri as the referer and to construct the answer uri.
   options.headers['Referer'] = uri.href;
   options.uri = uri.protocol + '//' + uri.hostname + '/cdn-cgi/l/chk_jschl';
+  // baseUrl can't be used in conjunction with an absolute uri
+  if (options.baseUrl !== undefined) {
```

Comment:
Why to have this if and not always set baseUrl to undentified?

Reply:
To avoid needlessly modifying state. The property is being assigned the value of undefined which causes the object to own that property. Why do that if It's not intended?

Reply:
To avoid needlessly modifying state. The property is being assigned the value of undefined which causes the object to own that property. Why do that if It's not intended?